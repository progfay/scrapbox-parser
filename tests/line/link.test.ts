/* global describe it expect */

describe('link', () => {
  it('Simple absolute link', () => {
    expect('https://example.com/').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'link',
            pathType: 'absolute',
            href: 'https://example.com/',
            content: ''
          }
        ]
      }
    ])
  })

  it('Simple absolute link with bracket', () => {
    expect('[https://example.com/]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'link',
            pathType: 'absolute',
            href: 'https://example.com/',
            content: ''
          }
        ]
      }
    ])
  })

  it('Simple root link', () => {
    expect('[/project/page]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'link',
            pathType: 'root',
            href: '/project/page',
            content: ''
          }
        ]
      }
    ])
  })

  it('Simple relative link', () => {
    expect('[page]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'link',
            pathType: 'relative',
            href: 'page',
            content: ''
          }
        ]
      }
    ])
  })

  it('Link with content', () => {
    expect(`[https://example.com/   Example]
[Example   https://example.com/]
[https://left.com/ center https://right.com/]`).toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'link',
            pathType: 'absolute',
            href: 'https://example.com/',
            content: 'Example'
          }
        ]
      },
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'link',
            pathType: 'absolute',
            href: 'https://example.com/',
            content: 'Example'
          }
        ]
      },
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'link',
            pathType: 'absolute',
            href: 'https://left.com/',
            content: 'center https://right.com/'
          }
        ]
      }
    ])
  })

  it('Root and relative link path can include space', () => {
    expect(`[page name]
[/project/page name]`).toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'link',
            pathType: 'relative',
            href: 'page name',
            content: ''
          }
        ]
      },
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'link',
            pathType: 'root',
            href: '/project/page name',
            content: ''
          }
        ]
      }
    ])
  })

  it('Link with link', () => {
    expect('[https://example.com https://example.com]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'link',
            pathType: 'absolute',
            href: 'https://example.com',
            content: 'https://example.com'
          }
        ]
      }
    ])
  })
})
