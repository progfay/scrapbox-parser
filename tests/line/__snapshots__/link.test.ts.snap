// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link Link with GET parameters 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "",
        "href": "http://example.com?key1=value1&key2=value2",
        "pathType": "absolute",
        "raw": "[http://example.com?key1=value1&key2=value2]",
        "type": "link",
      },
    ],
    "type": "line",
  },
]
`;

exports[`link Link with content 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "Example",
        "href": "https://example.com/",
        "pathType": "absolute",
        "raw": "[https://example.com/   Example]",
        "type": "link",
      },
    ],
    "type": "line",
  },
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "Example",
        "href": "https://example.com/",
        "pathType": "absolute",
        "raw": "[Example   https://example.com/]",
        "type": "link",
      },
    ],
    "type": "line",
  },
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "center https://right.com/",
        "href": "https://left.com/",
        "pathType": "absolute",
        "raw": "[https://left.com/ center https://right.com/]",
        "type": "link",
      },
    ],
    "type": "line",
  },
]
`;

exports[`link Link with link 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "https://example.com",
        "href": "https://example.com",
        "pathType": "absolute",
        "raw": "[https://example.com https://example.com]",
        "type": "link",
      },
    ],
    "type": "line",
  },
]
`;

exports[`link Root and relative link path can include space 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "",
        "href": "page name",
        "pathType": "relative",
        "raw": "[page name]",
        "type": "link",
      },
    ],
    "type": "line",
  },
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "",
        "href": "/project/page name",
        "pathType": "root",
        "raw": "[/project/page name]",
        "type": "link",
      },
    ],
    "type": "line",
  },
]
`;

exports[`link Simple absolute link 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "",
        "href": "https://example.com/",
        "pathType": "absolute",
        "raw": "https://example.com/",
        "type": "link",
      },
    ],
    "type": "line",
  },
]
`;

exports[`link Simple absolute link with ahead non-space character 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "raw": "a",
        "text": "a",
        "type": "plain",
      },
      Object {
        "content": "",
        "href": "https://example.com/",
        "pathType": "absolute",
        "raw": "https://example.com/",
        "type": "link",
      },
    ],
    "type": "line",
  },
]
`;

exports[`link Simple absolute link with bracket 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "",
        "href": "https://example.com/",
        "pathType": "absolute",
        "raw": "[https://example.com/]",
        "type": "link",
      },
    ],
    "type": "line",
  },
]
`;

exports[`link Simple relative link 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "",
        "href": "page",
        "pathType": "relative",
        "raw": "[page]",
        "type": "link",
      },
    ],
    "type": "line",
  },
]
`;

exports[`link Simple root link 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "",
        "href": "/project/page",
        "pathType": "root",
        "raw": "[/project/page]",
        "type": "link",
      },
    ],
    "type": "line",
  },
]
`;
